### PR TITLE
Another series of fixes for STM32MP13 clock driver

### PR DESCRIPTION
**Pull request by etienne-lms**

### CODE REVIEW

**Comment by etienne-lms** [comment]
I've updated the commit message CI / Code style test complained about.

**Comment by GseoC** [comment]
For commits:
`drivers: clk: clk-stm32mp13: add ADC and SPI clocks`
and
`drivers: clk: clk-stm32mp13: round up VCO to the nearest frequency`

`Reviewed-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Comment by etienne-lms** [comment]
Review tag still missing for commit "drivers: clk: clk-stm32mp13: don't gate/ungate oscillators not wired".

**Comment by GseoC** [comment]
> Review tag still missing for commit "drivers: clk: clk-stm32mp13: don't gate/ungate oscillators not wired".

There's still an unanswered question above, I'll be happy to give my tag afterwards :)

**Comment by etienne-lms** [comment]
Sorry, I can't see the question you're referring to.

File: core/drivers/clk/clk-stm32mp13.c [hunk 1]
```diff
@@ -1970,11 +1970,50 @@ const struct clk_ops ck_timer_ops = {
...
+
+#define STM32_OSCILLATOR(_name, _parent, _flags, _gate_id)\
+	struct clk _name = {\
+		.ops = &clk_stm32_oscillator_ops,\
+		.priv = &(struct clk_stm32_gate_cfg) {\
+			.gate_id = (_gate_id),\
+		},\
+		.name = #_name,\
+		.flags = (_flags),\
+		.num_parents = 1,\
```
Comment (etienne-lms) [inline core/drivers/clk/clk-stm32mp13.c:2008]: I should fix this. It's odd `num_parent` is 1 while `.parents` is null :(
I'll fix that.

**Comment by etienne-lms** [comment]
Issue in `STM32_OSCILLATOR()` macro fixed in appended fixup commit.

**Comment by etienne-lms** [comment]
Rebased for CI test purpose.

**Comment by etienne-lms** [comment]
Rebased for CI test purpose.

File: core/drivers/clk/clk-stm32mp13.c [hunk 2]
```diff
@@ -1487,7 +1487,7 @@ static int stm32_clk_parse_fdt_all_oscillator(const void *fdt,
 		fdt_err = clk_stm32_parse_oscillator_fdt(fdt, osc_node,
 							 osc_data->name, osci);
 		if (fdt_err < 0)
```
Comment (GseoC) [inline core/drivers/clk/clk-stm32mp13.c:1489]: I think it should still panic if <0 and != FDT_ERR_NOTFOUND

Comment (GseoC) [reply core/drivers/clk/clk-stm32mp13.c:1489]: Oh at least return the error code :)

Comment (etienne-lms) [reply core/drivers/clk/clk-stm32mp13.c:1489]: right, i'll panic if case of an unexpected error code.

**Comment by etienne-lms** [comment]
comment addressed.

**Comment by GseoC** [comment]
`Reviewed-by: Gatien Chevallier <gatien.chevallier@foss.st.com>`

**Pull request merged**